### PR TITLE
Modernize code: very last fixes

### PR DESCRIPTION
**Pull request by nerzhul**
It's the very last fixes

### CODE REVIEW

File: src/fontengine.cpp [hunk 1]
```diff
@@ -113,15 +113,15 @@ FontEngine::~FontEngine()
...
 void FontEngine::cleanCache()
 {
-	for ( unsigned int i = 0; i < FM_MaxMode; i++) {
+	for (auto &i : m_font_cache) {
 
 		for (std::map<unsigned int, irr::gui::IGUIFont*>::iterator iter
-				= m_font_cache[i].begin();
-				iter != m_font_cache[i].end(); ++iter) {
+				= i.begin();
+				iter != i.end(); ++iter) {
```
Comment (SmallJoker) [inline src/fontengine.cpp]: Range based for possible

File: src/guiFormSpecMenu.cpp [hunk 2]
```diff
@@ -113,8 +113,8 @@ GUIFormSpecMenu::~GUIFormSpecMenu()
 {
 	removeChildren();
 
-	for (u32 i = 0; i < m_tables.size(); ++i) {
-		GUITable *table = m_tables[i].second;
+	for (auto &table_it : m_tables) {
+		GUITable *table = table_it.second;
```
Comment (SmallJoker) [inline src/guiFormSpecMenu.cpp]: `table_it.second->drop()`

File: src/guiFormSpecMenu.cpp [hunk 3]
```diff
@@ -1958,8 +1951,8 @@ void GUIFormSpecMenu::regenerateGui(v2u32 screensize)
 	// Remove children
 	removeChildren();
 
-	for (u32 i = 0; i < m_tables.size(); ++i) {
-		GUITable *table = m_tables[i].second;
+	for (auto &table_id : m_tables) {
+		GUITable *table = table_id.second;
```
Comment (SmallJoker) [inline src/guiFormSpecMenu.cpp]: `table_id.second->drop()`

File: src/guiKeyChangeMenu.cpp [hunk 4]
```diff
@@ -105,15 +104,12 @@ void GUIKeyChangeMenu::removeChildren()
 {
 	const core::list<gui::IGUIElement*> &children = getChildren();
 	core::list<gui::IGUIElement*> children_copy;
-	for (core::list<gui::IGUIElement*>::ConstIterator i = children.begin(); i
-		 != children.end(); i++)
-	{
-		children_copy.push_back(*i);
+	for (gui::IGUIElement*i : children) {
```
Comment (SmallJoker) [inline src/guiKeyChangeMenu.cpp]: Space missing.

Comment (nerzhul) [inline src/guiKeyChangeMenu.cpp]: it's exact this cleaned code seems to show we are doing a useless copy

Comment (nerzhul) [inline src/guiKeyChangeMenu.cpp]: in fact it's needed by irrlicht, else we have a coredump

Comment (nerzhul) [inline src/guiKeyChangeMenu.cpp]: copy not possible: "error: no type named 'value_type' in 'irr::core::list<irr::gui::IGUIElement *>'"

**Comment by nerzhul** [comment]
rebased with macosx changes

**Pull request merged**